### ISSUE TITLE
Update documentation and checklists for ES6 modules

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/chipper#854]
From the doc:

(CM, ? hours) [post-publication] Update documentation and checklists.  Anything that refers to requirejs - PhET Developer Guide, code examples, checklists, source-code comments,...

We'll want to do this presumably AFTER we've migrated and committed.

**Comment by pixelzoom** [comment]
Deferred until ES6 migration has occurred.

**Comment by pixelzoom** [comment]
List of things to change:

- [ ] [Module](https://github.com/phetsims/phet-info/blob/master/doc/phet-software-design-patterns.md#module) section other require.js references of PhET Software Design Patterns
- [ ] [Modularity and RequireJS](https://github.com/phetsims/phet-info/blob/master/doc/phet-development-overview.md#modularity-and-requirejs) section and other require.js references in PhET Development Overview

**Comment by pixelzoom** [comment]
Before I can do any updates, we need to decide what "requirejs mode" should now be called.  Suggestions?

**Comment by pixelzoom** [comment]
In dev meeting document, @jbphet suggested:

Compilation free
Unbuilt
Uncompiled
Multi-file

**Comment by pixelzoom** [comment]
+1 for "built" and "unbuilt".

**Comment by pixelzoom** [comment]
I posted in Slack:

ALL DEVS: Proposal to used "built" and "unbuilt" to refer to what we previous called "built" and "requirejs" , see https://github.com/phetsims/chipper/issues/854#issuecomment-594930429.   This currently has +3 votes (including me).  Please weigh in (in the issue) by Monday 5/9 @ 5pm.   I can't proceed with revising docs and checklists until we have consensus.

**Comment by pixelzoom** [comment]
https://github.com/phetsims/chipper/issues/854#issuecomment-589779806 is a reasonable checklist of what needs to be done.  And updating the "Design Patterns" doc has been divided up by section author.  I'm way behind on higher priority work.  Could we divvy up the remaining checklist items?

Assigning to @ariel-phet to divvy up and prioritize.

**Comment by ariel-phet** [comment]
I will request volunteers at dev-meeting for items in https://github.com/phetsims/chipper/issues/854#issuecomment-589779806

**Comment by brandonLi8** [comment]
I took a stab at [Modularity with ES6 Modules](https://github.com/phetsims/phet-info/blob/master/doc/phet-development-overview.md#modularity-with-es6-modules) in `phet-development-overview.md`.  I choose to not mention commonJS repos (like chipper) as it is not directly ran in simulation code.

**Comment by jessegreenberg** [comment]
@jessegreenberg can take a look at the remaining items in https://github.com/phetsims/chipper/issues/854#issuecomment-589779806

**Comment by jessegreenberg** [comment]
> third-party-licenses.md

These will be done in #905. The rest of the items in https://github.com/phetsims/chipper/issues/854#issuecomment-589779806 have been done.  Closing this issue.
